Objects/odictobject.c

```diff
@@ -953,8 +953,9 @@ OrderedDict_fromkeys_impl(PyTypeObject *type, PyObject *seq, PyObject *value)
 PyDoc_STRVAR(odict_sizeof__doc__, "");
 
 static PyObject *
-odict_sizeof(PyODictObject *od, PyObject *Py_UNUSED(ignored))
+odict_sizeof(PyObject *op, PyObject *Py_UNUSED(ignored))
 {
+    PyODictObject *od = (PyODictObject*)op;
```

Comment:
Do you want a macro / static inline function for the cast so that you can also assert the type of `op` (as it's a static type, it's easier to do so since we don't need a module state)